src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -94,6 +105,11 @@ internal override void Validate()
                 countOfCredentialTypesSpecified++;
             }
 
+            if (Config.ClientAssertion != null)
+            {
+                countOfCredentialTypesSpecified++;
+            }
+
             if (countOfCredentialTypesSpecified > 1)
             {
                 throw new InvalidOperationException(MsalErrorMessage.ClientSecretAndCertificateAreMutuallyExclusive);
```

Comment:
MSAL should not throw InvalidOperationException when validating params, but MsalClientException. 
Also, you should update the error message as there are now 3 options that are mutually exclusive.

Reply:
Can this be changed to a MsalClientException?

---
In reply to: [292452633](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1208#discussion_r292452633) [](ancestors = 292452633)

Reply:
Yes, ill make the change